### PR TITLE
Use 'cwd' and 'program' arguments only if they exist

### LINKED ISSUE #642: debugpy Default Config returns Invalid Message
Latest `dap-mode` (Melpa 20220712.919) with debugpy 1.6.2 is returning the following on `dap-debug`, choosing default templates returns:
```
Warning (emacs): Initialize request failed: Invalid message: "program" must be str
```
for `buffer` and `pointer` templates, and:
```
Warning (emacs): Initialize request failed: Invalid message: "cwd" must be str
```

for `file`.

I'm not sure if this is a configuration issue on my end, but here is relevant config:
```
(use-package dap-mode
  :straight t
  :custom
  (dap-auto-configure-features '(sessions locals controls tooltip))
  (dap-python-debugger 'debugpy)
  :config
  (require 'dap-python))
```
stepping through edebugger shows `launch-args` evaluating with:
```
Result: (:type "python" :module "pytest" :request "launch" :name "Python :: Run pytest (buffer)" :args ("/Users/chrismoore/Projects/...") :dap-server-path ("/Users/chrismoore/.asdf/shims/python" "-m" "debugpy.adapter"))
```

Are default templates not expected to work? I see the project still defaults to `ptvsd` (which runs as expected)

### PR DESCRIPTION
**Pull request by factyy**
Probably fix #642

### CODE REVIEW

File: dap-mode.el [hunk 1]
```diff
@@ -1738,8 +1738,10 @@ before starting the debug process."
 
               (dap--set-sessions (cons debug-session debug-sessions)))
             (let ((translated-launch-args (cl-copy-list launch-args)))
-              (plist-put translated-launch-args :cwd (funcall (dap--debug-session-local-to-remote-path-fn debug-session) cwd))
-              (plist-put translated-launch-args :program (funcall (dap--debug-session-local-to-remote-path-fn debug-session) program))
+              (if cwd
```
Comment (s-kostyaev) [inline dap-mode.el:1741]: better to use `when` instead of if without else branch

Comment (factyy) [reply dap-mode.el]: Fixed in 6ce259f

**Review by s-kostyaev — APPROVED** [review]
LGTM

**Review by s-kostyaev — APPROVED** [review]
LGTM

**Review by yyoncho — APPROVED** [review]

**Pull request merged**